mesh/mesh.cpp

```diff
@@ -1552,6 +1552,153 @@ Geometry::Type Mesh::GetTypicalElementGeometry() const
    return geom;
 }
 
+void Mesh::GetExteriorFaceMarker(Array<int> & face_marker) const
+{
+   if (face_marker.Size() < faces_info.Size())
+   {
+      face_marker.SetSize(faces_info.Size());
+   }
```

Comment:
If the output array is bigger than the number of faces, I think it is best to shrink its size to be the number of faces. In other words -- simply call `face_marker.SetSize(num_faces)` where `num_faces = GetNumFaces()`.